src/NServiceBus.Transport.Bridge/MessageShovel.cs

```diff
@@ -29,14 +29,29 @@ public async Task TransferMessage(TransferContext transferContext, CancellationT
 
             var transferDetails = $"{transferContext.SourceTransport}->{targetEndpointDispatcher.TransportName}";
 
-            // Audit messages contain all the original fields. Transforming them would destroy this.  
-            if (!IsAuditMessage(messageToSend))
+            if (IsErrorMessage(messageToSend))
             {
-                messageToSend.Headers[BridgeHeaders.Transfer] = transferDetails;
+                //This is a failed message forwarded to ServiceControl. We need to transform the FailedQ header so that ServiceControl returns the message
+                //to the correct queue/transport on the other side
 
-                TransformAddressHeader(messageToSend, targetEndpointRegistry, Headers.ReplyToAddress);
+                //We _do not_ transform the ReplyToAddress header
                 TransformAddressHeader(messageToSend, targetEndpointRegistry, FaultsHeaderKeys.FailedQ);
```

Comment:
I am worried that the test does not verify if

```
TransformAddressHeader(messageToSend, targetEndpointRegistry, FaultsHeaderKeys.FailedQ);
```

is executed because if we don't transform it, the address is left in the form of acceptance testing transform (`Retry.ProcessingEndpoint`) which is a valid address in all transports so it can be retried from SC.

Reply:
This is no longer relevant because the appropriate test has been added.